src/diff.ts

```diff
@@ -208,8 +207,25 @@ export async function upsertGitHubComment(
     comment.body?.includes(DIFF_COMMENT_IDENTIFIER)
   )
 
+  const emptyDiff = diff.trim() === ''
+
   // If we can find the comment we update it, otherwise we create a new comment
   if (comment) {
+    if (emptyDiff) {
+      // If the diff is empty, we delete the comment.
+      const deletedComment = await oktokit.rest.issues.deleteComment({
+        ...context.repo,
+        comment_id: comment.id
+      })
+      if (deletedComment.status !== 204) {
+        throw new Error('Failed to delete comment')
+      }
```

Comment:
What could cause the failure while deleting the comment?  

I wonder if it makes sense to log the error instead of failing the job.

Reply:
If the deletion fails it will throw an error and it will be captured in `main.tsx` with
```
} catch (error) {
    // Fail the workflow run if an error occurs
    if (error instanceof Error) core.setFailed(error.message)
  }
```

the check there is just for sanity reasons. It will thrown an error but once again it will be matched and logged

Reply:
Fair, but i was wondering should we fail the workflow if we fail to delete the comment. But properly makes sense to be more explicit and just silently logging the error.

Reply:
The workflow fails since we call `core.setFailed(error.message)`